lib/external_service_new/mailchimp_downloader.rb

```diff
@@ -20,50 +25,24 @@ def each_list(&block)
     end
 
     def each_email(id:, filters: nil, &block)
-      MailchimpInternalDownloader.new(dispatcher: dispatcher, id: id, logger: logger).each_email(&block)
-    end
-
-    private
-
-    attr_reader :logger, :dispatcher
-  end
-
-  class MailchimpInternalDownloader
-    def initialize(dispatcher:, id:, logger:nil)
-      raise "dispatcher must not be nil, found #{dispatcher.inspect}" unless dispatcher
-
-      @dispatcher = dispatcher
-      @id         = id
-      @logger     = logger || respond_to?(:logger) ? logger : nil
-    end
-
-    def each_email(&block)
       return to_enum(:each_email) unless block
 
       mailing_list_infos = dispatcher.dispatch(:get, "lists/#{id}")
 
       offset = 0
       while offset < mailing_list_infos["stats"]["member_count"] do
         response = dispatcher.dispatch(:get, "lists/#{id}/members?count=1000&offset=#{offset}&fields[]=email_address,stats")
-        members = response["members"]
-
-        members.each {|member| yield member["email_address"]}
+        break if response["members"].empty?
 
-        if members.empty? then
-          logger && logger.warn(<<-EOF)
-            The members count we received is either empty (it should not!) or Mailchimp is bugged.
-            Either way, converting whatever emails we have retrieved yet...
-          EOF
+        response = ExternalServiceNew::MailchimpListMembersResponse.new(response["members"])
```

Comment:
Not sure that you need to prefix ExternalServiceNew since we already are in the module.